apps/studio/components/interfaces/ProjectCreation/RegionSelector.tsx

```diff
@@ -0,0 +1,62 @@
+import { CloudProvider, PROVIDERS } from 'lib/constants'
+import { useRouter } from 'next/router'
+import { useEffect } from 'react'
+import { Listbox } from 'ui'
+
+import { useDefaultRegionQuery } from 'data/misc/get-default-region-query'
+import { getAvailableRegions } from './ProjectCreation.utils'
+
+interface RegionSelectorProps {
+  cloudProvider: CloudProvider
+  selectedRegion: string
+  onSelectRegion: (value: string) => void
+}
+
+export const RegionSelector = ({
+  cloudProvider,
+  selectedRegion,
+  onSelectRegion,
+}: RegionSelectorProps) => {
+  const router = useRouter()
+  const availableRegions = getAvailableRegions(PROVIDERS[cloudProvider].id)
+  const { data: region, isLoading, isSuccess } = useDefaultRegionQuery({ cloudProvider })
+
+  useEffect(() => {
+    if (isSuccess && region) onSelectRegion(region)
+  }, [isSuccess])
```

Comment:
Shouldn't actually make a difference since `isSuccess` and `region` and linked, but better to be explicit
```suggestion
  }, [isSuccess, region])
```